test/e2e/user_group_limit/user_group_limit_test.go

```diff
@@ -0,0 +1,174 @@
+})
+
+var _ = ginkgo.Describe("UserGroupLimit", func() {
+	ginkgo.It("Verify_maxapplications", func() {
+		// update config
+		ginkgo.By("Update root.sandbox1 with 1 maxapplications and 1Gi max memory")
+		annotation = "ann-" + common.RandSeq(10)
+		yunikorn.UpdateCustomConfigMapWrapper(oldConfigMap, "", annotation, func(sc *configs.SchedulerConfig) error {
+			// remove placement rules so we can control queue
+			sc.Partitions[0].PlacementRules = nil
+
+			var err error
+			if err = common.AddQueue(sc, "default", "root", configs.QueueConfig{
+				Name: "sandbox1",
+				Limits: []configs.Limit{
+					{
+						Limit:           "Test entry",
+						Users:           []string{"kubernetes-admin"},
+						MaxApplications: 1,
+						MaxResources: map[string]string{
+							siCommon.Memory: "1Gi",
+						},
+					},
+				},
+			}); err != nil {
+				return err
+			}
+			return nil
+		})
+
+		// Deploy sleepPod1
+		sleepPodObjs := createSleepPodObjs(2, map[string]string{"queue": "root.sandbox1"})
+		sleepPodObj1 := sleepPodObjs[0]
+		ginkgo.By("Deploy the sleep pod " + sleepPodObj1.Name + " to the development namespace")
+		sleepPod1, podErr := kClient.CreatePod(sleepPodObj1, dev)
+		gomega.Ω(podErr).NotTo(gomega.HaveOccurred())
+		sleepPod1Application := sleepPod1.Labels[constants.LabelApplicationID]
+
+		// Wait for sleepPod1 to move to running state
+		podErr = kClient.WaitForPodBySelectorRunning(dev,
+			fmt.Sprintf("app=%s", sleepPod1.ObjectMeta.Labels["app"]),
+			60)
+		gomega.Ω(podErr).NotTo(gomega.HaveOccurred())
+
+		// Deploy sleepPod2
+		sleepPodObj2 := sleepPodObjs[1]
+		ginkgo.By("Deploy the sleep pod " + sleepPodObj2.Name + " to the development namespace")
+		sleepPod2, podErr := kClient.CreatePod(sleepPodObj2, dev)
+		gomega.Ω(podErr).NotTo(gomega.HaveOccurred())
+		sleepPod2Application := sleepPod2.Labels[constants.LabelApplicationID]
+
+		userUsageDAOInfo, err := restClient.GetUserUsage(constants.DefaultPartition, "kubernetes-admin")
+		Ω(err).NotTo(gomega.HaveOccurred())
+		Ω(userUsageDAOInfo).NotTo(gomega.BeNil())
+		Ω(len(userUsageDAOInfo.Queues.Children)).To(gomega.Equal(1))
+
+		sandboxResourceUsageDAO := userUsageDAOInfo.Queues.Children[0]
+		Ω(sandboxResourceUsageDAO.QueuePath).To(gomega.Equal("root.sandbox1"))
+		Ω(sandboxResourceUsageDAO.ResourceUsage).NotTo(gomega.BeNil())
+		Ω(sandboxResourceUsageDAO.ResourceUsage.Resources["pods"]).To(gomega.Equal(resources.Quantity(1)))
+		Ω(len(sandboxResourceUsageDAO.RunningApplications)).To(gomega.Equal(1))
+		Ω(sandboxResourceUsageDAO.RunningApplications[0]).To(gomega.Equal(sleepPod1Application))
+		Ω(sandboxResourceUsageDAO.RunningApplications[0]).NotTo(gomega.Equal(sleepPod2Application))
+
+		// // sleepPod2 can't be scheduled because application count is more than maxapplications
+		// ginkgo.By("The sleep pod " + sleepPod2.Name + " can't be scheduled")
+		// err = kClient.WaitForPodUnschedulable(sleepPod2, 60*time.Second)
+		// gomega.Ω(err).NotTo(gomega.HaveOccurred())
```

Comment:
Not sure whether I understand User Based Quota Enforcement correctly. If the `maxapplications` is 1, I expect a user can't create more than one application in a queue. However, the second application can be scheduled, but the related resource usage is not tracked in the user tracker. From the source code, it looks like we don't handle the result from [IncreaseTrackedResource](https://github.com/apache/yunikorn-core/blob/0ffd1392c329b185bdd8e51665d48e0502807892/pkg/scheduler/objects/application.go#L1656).

@wilfred-s, may you give me some feedback about this? Thank you!